libraries/AP_GPS/AP_GPS.h

```diff
@@ -34,8 +34,6 @@
 // the number of GPS leap seconds
 #define GPS_LEAPSECONDS_MILLIS 18000ULL
 
-#define UNIX_OFFSET_MSEC (17000ULL * 86400ULL + 52 * 10 * MSEC_PER_WEEK - GPS_LEAPSECONDS_MILLIS)
```

Comment:
@muramura Just tested your code in a PH2.1, it overflows. Two points:
- you have to make sure long is 32-bit in whatever platform you compiled; you also need to make sure you are using the same compiler version we do for our boards, it's possible different versions work differently
- I'm not aware of anywhere in the C++ spec that says it will cast to the largest type in the full math expression; it will cast to the largest of the two operands in an math operation, not the entire math expression. That means that `52 * 10` has no cast in either of the operands and then `520 * MSEC_PER_WEEK` has only 520 casted to unsigned long - the overflow happens then.